main_page.php

```diff
@@ -1,11 +1,27 @@
 <?php
-$db = new PDO('mysql:host=db; dbname=collectorapp', 'root', 'password');
-$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    /**
+    * Creates a connection to the collectorapp database
+    */
+    function createDBConnection()
```

Comment:
This one can be return type hinted as : PDO

Reply:
Make sure the function returns $db otherwise you wont actually be able to access it outside the function